### PR TITLE
Add RBAC rules for all resources

### PR DESCRIPTION
**Pull request by ggaurav10**

### CODE REVIEW

File: config/rbac/rbac_role.yaml [hunk 1]
```diff
@@ -28,6 +28,66 @@ rules:
...
+  verbs:
+  - get
+  - list
+  - watch
+  - update
+  - patch
+- apiGroups:
+  - apps
+  resources:
+  - daemonsets
```
Comment (amshuman-kr) [inline config/rbac/rbac_role.yaml]: Why not combine the `apps` group resources (`daemonsets`, `replicasets`, `replicationcontrollers`, `deployments` and `statefulsets` in one entry?

**Review by amshuman-kr — CHANGES_REQUESTED** [review]
Why not combine the apps group resources (daemonsets, replicasets, replicationcontrollers, deployments and statefulsets in one entry?

File: config/rbac/rbac_role.yaml (hunk 1 above)
Comment (ggaurav10) [reply config/rbac/rbac_role.yaml]: this file is autogenerated by kubebuilder when [these](https://github.com/gardener/hvpa-controller/pull/5/files#diff-bfa21ee442000583bc7aa6f53ec3660cR309) lines are added, and `make` is run.

Comment (ggaurav10) [reply config/rbac/rbac_role.yaml]: Done

**Review by amshuman-kr — APPROVED** [review]
Thanks for the changes. LGTM

**Pull request merged**